shared/fs/container.js

```diff
@@ -12,28 +12,46 @@ type OwnProps = {
 
 type StateProps = {
   path: Types.Path,
-  items: I.List<Types.Path>,
+  items: I.List<Types.PathItem>,
   progress: Types.ProgressType,
+  sortSetting: Types.SortSetting,
 }
 
-type DispatchProps = {}
+type DispatchProps = {
+  _sortSettingChangeFactory: (path: Types.Path) => (setting: Types._SortSetting) => void,
+}
 
 const mapStateToProps = (state: TypedState, {routeProps}: OwnProps) => {
   const path = Types.stringToPath(routeProps.get('path', Constants.defaultPath))
+  const sortSetting = state.fs.pathUserSettings.get(path, Constants.makePathUserSetting()).get('sort')
   const itemDetail = state.fs.pathItems.get(path)
-  const items = itemDetail && itemDetail.type === 'folder' ? itemDetail.get('children', I.List()) : I.List()
+  const itemNames =
+    itemDetail && itemDetail.type === 'folder' ? itemDetail.get('children', I.List()) : I.List()
+  const items: I.List<Types.PathItem> = itemNames
```

Comment:
same as this, don't do the mapping and concatting here

Reply:
Got it. But this mapping needs to access `state.fs.pathItems`, which isn't available in `mergeProps`. Should I pass `pathItems` into `mergeProps` as well?

Reply:
Hmm I guess that makes more sense. Will do!